tests/memory/enc/boundaries.c

```diff
@@ -17,16 +18,27 @@ void test_host_boundaries(buffer buf)
 
 void test_enclave_boundaries()
 {
-    void* array[ITERS];
+    void* array[ITERS] = {NULL};
+    /* Calculate the upper bound of the heap. */
+    size_t bound = __oe_get_heap_size() - OE_USED_HEAP;
+    size_t allocated = 0;
+
     for (int i = 0; i < ITERS; i++)
     {
+        if (allocated >= bound)
```

Comment:
Is this equivalent to just checking if malloc returns NULL and if so, break?

Reply:
The test (below) uses the OE_TEST to test the return of malloc, so for the consistency, I add the check against the upper bound here.